packages/uni_ui/lib/calendar/calendar_item.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:figma_squircle/figma_squircle.dart';
+import 'package:flutter/material.dart';
+
+class _CalendarItemDate extends StatelessWidget {
+  const _CalendarItemDate({
+    required this.eventPeriod,
+  });
+
+  final DateTimeRange? eventPeriod;
+
+  @override
+  Widget build(BuildContext context) {
+    if (eventPeriod != null) {
+      return Column(
+        children: [
+          Text(
+            parsePeriod(eventPeriod!),
+            style: TextStyle(
+              fontSize: 15,
+              height: 1,
+              fontWeight: FontWeight.w500,
+            ),
+          ),
+          Text(
+            eventPeriod!.end.year.toString(),
+            style: TextStyle(
+              color: Theme.of(context).colorScheme.outline,
+              fontSize: 11,
+            ),
+          ),
+        ],
+      );
+    } else {
+      return Padding(
+        padding: EdgeInsets.only(top: 10),
+        child: Text(
+          "TBD",
+          style: TextStyle(
+            fontSize: 15,
+            fontWeight: FontWeight.w500,
+          ),
+        ),
+      );
+    }
+  }
+
+  static String monthToString(int month) {
+    // TODO: Support Portuguese
+    const strMonths = [
```

Comment:
See #1379, I added a datetime extension method for this case. However, I think this parsing should be moved to uni_app, and just receive a string. What do you think?